android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -169,64 +169,64 @@ protected void performUpdate() {
-            break;
-          case SLIDE_FROM_LEFT:
-            customAnimation = true;
-            getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
-            break;
-        }
-
-        if (!customAnimation) {
-          getOrCreateTransaction().setTransition(transition);
-        }
+        // if the previous top screen does not exist anymore and the new top was not on the stack before,
+        // probably replace or reset was called, so we play the "close animation"
+        // otherwise it's open animation
+        shouldUseOpenAnimation = mScreenFragments.contains(mTopScreen) || newTop.getScreen().getReplaceAnimation() != Screen.ReplaceAnimation.POP;
+        stackAnimation = newTop.getScreen().getStackAnimation();
       }
     } else if (mTopScreen != null && !mTopScreen.equals(newTop)) {
-      // otherwise if we are performing top screen change we do "back animation"
-      int transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
+      // otherwise if we are performing top screen change we do "close animation"
+      shouldUseOpenAnimation = false;
+      stackAnimation = mTopScreen.getScreen().getStackAnimation();
+    }
 
-      switch (mTopScreen.getScreen().getStackAnimation()) {
-        case NONE:
-          transition = FragmentTransaction.TRANSIT_NONE;
+    // animation logic start
+    if (shouldUseOpenAnimation) {
+      transition = FragmentTransaction.TRANSIT_FRAGMENT_OPEN;
+
+      switch (stackAnimation) {
+        case SLIDE_FROM_RIGHT:
+          getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_right, R.anim.rns_slide_out_to_left);
           break;
-        case FADE:
-          transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
+        case SLIDE_FROM_LEFT:
+          getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
           break;
+      }
+    } else {
+      transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
+
+      switch (stackAnimation) {
         case SLIDE_FROM_RIGHT:
-          customAnimation = true;
           getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
           break;
         case SLIDE_FROM_LEFT:
-          customAnimation = true;
           getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_right, R.anim.rns_slide_out_to_left);
           break;
       }
+    }
 
-      if (!customAnimation) {
-        getOrCreateTransaction().setTransition(transition);
-      }
+    if (stackAnimation == Screen.StackAnimation.NONE) {
+      transition = FragmentTransaction.TRANSIT_NONE;
     }
+    if (stackAnimation == Screen.StackAnimation.FADE) {
+      transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
+    }
+
+    // check if it is a custom animation
```

Comment:
Is this comment necessary?

Reply:
Removed in 9722871393482a56e8f10e2a77dce2679fe8c23c